m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -20,53 +20,35 @@
 
 package com.uber.m3.tally.m3;
 
-import com.uber.m3.tally.BucketPair;
-import com.uber.m3.tally.BucketPairImpl;
-import com.uber.m3.tally.Buckets;
-import com.uber.m3.tally.Capabilities;
-import com.uber.m3.tally.CapableOf;
-import com.uber.m3.tally.StatsReporter;
+import com.uber.m3.tally.*;
 import com.uber.m3.tally.m3.thrift.TCalcTransport;
 import com.uber.m3.tally.m3.thrift.TMultiUdpClient;
 import com.uber.m3.tally.m3.thrift.TUdpClient;
 import com.uber.m3.tally.m3.thrift.TUdpTransport;
-import com.uber.m3.thrift.gen.CountValue;
-import com.uber.m3.thrift.gen.GaugeValue;
-import com.uber.m3.thrift.gen.M3;
-import com.uber.m3.thrift.gen.Metric;
-import com.uber.m3.thrift.gen.MetricBatch;
-import com.uber.m3.thrift.gen.MetricTag;
-import com.uber.m3.thrift.gen.MetricValue;
-import com.uber.m3.thrift.gen.TimerValue;
+import com.uber.m3.thrift.gen.*;
 import com.uber.m3.util.Duration;
 import com.uber.m3.util.ImmutableMap;
+import org.apache.http.annotation.NotThreadSafe;
+import org.apache.thrift.TBase;
 import org.apache.thrift.TException;
 import org.apache.thrift.protocol.TCompactProtocol;
 import org.apache.thrift.protocol.TProtocol;
 import org.apache.thrift.protocol.TProtocolFactory;
 import org.apache.thrift.transport.TTransport;
+import org.apache.thrift.transport.TTransportException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.net.InetAddress;
 import java.net.SocketAddress;
 import java.net.SocketException;
 import java.net.UnknownHostException;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.concurrent.BlockingQueue;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.LinkedBlockingQueue;
-import java.util.concurrent.Phaser;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
-
-import org.apache.thrift.transport.TTransportException;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
```

Comment:
not a fan of * imports, can you change and list out.

Reply:
SG